src/graphql/root/query/on-chain-tx-fee-query.ts

```diff
@@ -15,15 +15,16 @@ const OnChainTxFeeQuery = GT.Field({
     amount: { type: GT.NonNull(SatAmount) },
     targetConfirmations: { type: TargetConfirmations, defaultValue: 1 },
   },
-  resolve: async (_, args) => {
+  resolve: async (_, args, { domainAccount }: { domainAccount: Account }) => {
     const { walletId, address, amount, targetConfirmations } = args
 
     for (const input of [walletId, address, amount, targetConfirmations]) {
       if (input instanceof Error) throw input
     }
 
-    const fee = await Wallets.getOnChainFeeByWalletId({
+    const fee = await Wallets.getOnChainFee({
       walletId,
+      withdrawFee: domainAccount.withdrawFee,
```

Comment:
I think this is an example of where we can pass in the fully formed `domainAccount` into the use case. Then the use case can also validate that the `wallet` has the correct `accountId`.

Reply:
this is actually what I have done initially, then I realize we only need the one value from the `Account` and this is the `withdrawFee`, so why having this use case depend on the whole `Account` when we just need a single value?

I have not thought about the validation, but almost by definition, it's been fetched from the `sessionContext`, so do we need validation?

Reply:
I think the "dummer" the resolvers the better. Knowing that an Account has a `withdrawFee` field is already more coupling than necessary.
Having an account in the context is a performance optimisation - it should just be passed along. The use cases / domain is where internal knowledge of the entities should be located.
Consider if we evolve the way we are representing fees. They could be dynamic (see liquidity leacher) for example. At that point a change in the domain would require a change to this resolver. But we know that fees will remain coupled to an account - so if we just pass the account then the change is only located to domain code (as it should be).

Finally we do need to validate that the user has access to the wallet. We consider the account valid when it comes from the context but we need to check that the wallet is actually one connected to the account.